cpp/src/unary/math_ops.cu

```diff
@@ -230,6 +230,59 @@ struct DeviceNot {
   }
 };
 
+template <typename T>
+struct fixed_point_ceil {
+  T n;
+  __device__ T operator()(T data)
+  {
+    T const a = (data / n) * n;                  // result of integer division
+    return a + (data > 0 && a != data ? n : 0);  // add 1 if positive and not round number
```

Comment:
This comment gave me the wrong initial impression that `T n` is always 1. I get that `n` is 1 scaled to the proper spot now, but I wonder if this is a bit dense and could use some more explanation. I would love to see some comments about these functors describing the operation and how it is accomplished. I think it would greatly help later when revisiting the code. Maybe something like:
```
/*
 * Ceiling is calculated using standard integer division. First we take the fixed_point value without regard to scale, 
 * divide by 1 scaled up to match the representation and then divide by that scaled one again. This results 
 * in truncating the value. For a negative number, this is all that is needed since the ceiling operation is 
 * defined as the least integer greater than the value. For a positive number, we may need to round it up. This 
 * is handled by comparing the truncated value to the original value and if they are not equal, the truncated 
 * value had a decimal component and the result needs to increment by the scaled 1.
 */
```

Reply:
Let me know if you think the comment I added (a tweaked version of yours) is sufficient.

Reply:
These are great and will really help people coming into this code to figure out what is happening. I like the additions you put for clarification on `n` as well. Thanks.